server.go

```diff
@@ -336,3 +342,17 @@ func (sb *SplitBytes) Next() bool {
 func (sb *SplitBytes) Chunk() []byte {
 	return sb.currentChunk
 }
+
+func (s *Server) registerPlugin(p plugin) {
+	s.pluginMtx.Lock()
+	s.plugins = append(s.plugins, p)
+	s.pluginMtx.Unlock()
```

Comment:
nitpick: can we defer unlock? likewise for getPlugins

I'm personally not really sure we need the mutex at all, since we only register plugins at the start and then the array is never mutated again, but I assume that's going to change and you added the mutex to prepare for that